apps/ledger-live-desktop/src/main/setup.ts

```diff
@@ -7,17 +7,44 @@ import fs from "fs/promises";
 import updater from "./updater";
 import resolveUserDataDirectory from "~/helpers/resolveUserDataDirectory";
 import path from "path";
+import { InMemoryLogger } from "./logger";
+
 ipcMain.on("mainCrashTest", () => {
   captureException(new Error("CrashTestMain"));
 });
+
 ipcMain.on("updater", (e, type) => {
   updater(type);
 });
+
+/**
+ * Saves logs from the renderer thread and logs recorded from the internal thread to a file.
+ */
 ipcMain.handle(
   "save-logs",
-  async (event, path: Electron.SaveDialogReturnValue, experimentalLogs: string) =>
-    !path.canceled && path.filePath && fs.writeFile(path.filePath, experimentalLogs),
+  async (_event, path: Electron.SaveDialogReturnValue, rendererLogs: unknown[]) => {
+    if (!path.canceled && path.filePath) {
+      const inMemoryLogger = InMemoryLogger.getLogger();
+      const internalLogs = inMemoryLogger.getLogs();
+      console.log(
+        `Saving ${rendererLogs.length} logs from the renderer thread and ${internalLogs.length} logs from the internal thread`,
+      );
+
+      // Merging according to a `date` (internal logs) / `timestamp` (most of renderer logs) does not seem necessary.
```

Comment:
the only tradeoff is we may miss context of things by doing so, like we can't understand the relation because user actions (renderer) and the internal process behavior.

do we at least still have a correct date set on the logs from internal?

Reply:
We have a correct data set on the logs from internal

But there is a possibility that a log coming from `renderer` has the same timestamp than a log coming from the `internal` thread: in that case, how do we decide which logs happened first ?

Reply:
i think it's ok for the time being, in future it wouldn't be too hard to sort by date if we want to, or even make it done on the logsviewer instead of inplace anyway. 
cheers